packages/xo-server/src/remote-handlers/local.js

```diff
@@ -68,7 +68,9 @@ export default class LocalHandler extends RemoteHandlerAbstract {
 
   async _createOutputStream (file, options) {
     const path = this._getFilePath(file)
-    await fs.ensureDir(dirname(path))
+    if (!options.fd) {
```

Comment:
IMHO it's weird to provide both a path and an fd, either one or the other should be provided:

```js
const file = 'foo.bar'
await remote.createOutputStream(file)

// or
const file = { fd: await remote.open('foo.bar') }
await remote.createOutputStream(file)
```

What do you think?

Reply:
I reproduced the fs.createReadStream() https://nodejs.org/dist/latest-v6.x/docs/api/fs.html#fs_fs_createreadstream_path_options weirdness without trying to improve on it. I suggest we give a completely different name and interface to the method if we improve on it.

Reply:
Yes, I've seen the doc, it's indeed pretty weird.

IMHO we can diverge from it on this aspect, it's already assumed that it's not perfectly compliant with Node's API.

Reply:
ok for the API change. 
I am a bit skeptical of having a client.createReadStream(file, option) where both file and options could be objects having properties clobbing each other, I would almost spell out the arguments explicitly and 2 methods:``` client.createReadStream(filePath, flags, start, end)``` and ```client.createReadStreamFromFd(fd, start, end)```

Reply:
I understand your concern but I think it's going to far for what we need right now, let's keep the changes minimal, just add the fd support :-)

Reply:
ok, will do tomorrow

Reply:
```
  async _createOutputStream (file, options) {
    if (typeof file === 'string') {
      const path = this._getFilePath(file)
      await fs.ensureDir(dirname(path))
      return fs.createWriteStream(path, options)
    } else {
      return fs.createWriteStream('', {...options, fd: file.fd, autoClose: false})
    }
  }
```
is that what you had in mind?

Reply:
Yes, that seems easy to use, almost a drop-in replacement for the current uses :)

Reply:
ok, I will make the change